src/riscv/lib/src/machine_state/memory.rs

```diff
@@ -307,6 +317,33 @@ pub trait MemoryConfig: Send + 'static {
         F: FnManager<Ref<'a, M>>;
 }
 
+/// Trait that handles memory permission updates for the jit.
+/// The jit router needs to know when the memory permissions
+/// are updated so it can invalidate the jit instances.
```

Comment:
Let's not overload the description of this trait with details of the trait implementations.

```suggestion
/// Implementing types will be informed about changes to memory permissions
```